akka-javasdk/src/main/java/akka/javasdk/http/AbstractHttpEndpoint.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.http;
+
+import akka.annotation.InternalApi;
+
+/**
+ * Optional base class for HTTP endpoints giving access to a request context without additional constructor parameters
+ */
+abstract public class AbstractHttpEndpoint {
```

Comment:
Alternatively, we could call it `RequestContextSupport`. 

And maybe better if we make this a interface with default methods. As such users can use it as a mixin instead. 

And have some other base classes on their own.

Reply:
We need a field so interface does not work.

Reply:
Oh yes, that's true.